packages/components/src/FilterDateRangePicker/subcomponents/FilterDateRangePickerField/FilterDateRangePickerField.tsx

```diff
@@ -0,0 +1,248 @@
+import { useDateInputHandlers } from "@kaizen/date-picker/src/hooks/useDateInputHandlers"
+import { calculateDisabledDays } from "@kaizen/date-picker/src/utils/calculateDisabledDays"
+import { formatDateAsText } from "@kaizen/date-picker/src/utils/formatDateAsText"
+import { getLocale } from "@kaizen/date-picker/src/utils/getLocale"
+import { isInvalidDate } from "@kaizen/date-picker/src/utils/isInvalidDate"
+import { parseDateFromTextFormatValue } from "@kaizen/date-picker/src/utils/parseDateFromTextFormatValue"
+import { DataAttributes } from "~types/DataAttributes"
+import { OverrideClassName } from "~types/OverrideClassName"
+import {
+  DateRangeInputField,
+  DateRangeInputFieldProps,
+} from "../DateRangeInputField"
+import { useEndDateValidation } from "./hooks/useEndDateValidation"
+import { useStartDateValidation } from "./hooks/useStartDateValidation"
+import {
+  DateRange,
+  DateRangeFieldValidationMessage,
+  DateValidationResponse,
+  DisabledDayMatchers,
+  SupportedLocales,
+} from "./types"
+import styles from "./FilterDateRangePickerField.module.scss"
+
+type InputRangeStartProps = DateRangeInputFieldProps["inputRangeStartProps"]
+type InputRangeEndProps = DateRangeInputFieldProps["inputRangeEndProps"]
+
+type FilterInputProps<InputProps> = Omit<Partial<InputProps>, "value"> &
+  DataAttributes
+
+export interface FilterDateRangePickerFieldProps
+  extends OverrideClassName<HTMLAttributes<HTMLDivElement>>,
+    DisabledDayMatchers {
+  id: string
+  label: string
+  locale: SupportedLocales
+  /**
+   * Sets first displayed month to month of provided date if there isn't a date set.
+   */
+  defaultMonth?: CalendarRangeProps["defaultMonth"]
+  /**
+   * The date range passed in from the consumer that renders in the inputs and calendar.
+   */
+  selectedRange: DateRange | undefined
+  /**
+   * Callback when date is updated either by the calendar picker or by typing and blurring.
+   * Date will return as `undefined` if empty, invalid or disabled.
+   */
+  onRangeChange: (range: DateRange | undefined) => void
+  inputRangeStartProps?: FilterInputProps<InputRangeStartProps>
+  inputRangeEndProps?: FilterInputProps<InputRangeEndProps>
+  /**
+   * Custom description to provide extra context (input format help text remains).
+   */
+  description?: DateRangeInputFieldProps["description"]
+  validationMessage?: DateRangeFieldValidationMessage
+  /**
+   * Callback when a date is selected. Utilises internal validation if not set.
+   */
+  onValidate?: {
+    dateStart?: (validationResponse: DateValidationResponse) => void
+    dateEnd?: (validationResponse: DateValidationResponse) => void
+  }
+}
+
+export const FilterDateRangePickerField = ({
+  id,
+  label,
+  locale: propsLocale,
+  defaultMonth,
+  selectedRange,
+  onRangeChange,
+  disabledDates,
+  disabledDaysOfWeek,
+  disabledRange,
+  disabledBeforeAfter,
+  disabledBefore,
+  disabledAfter,
+  inputRangeStartProps,
+  inputRangeEndProps,
```

Comment:
Could these be `InputStartDateProps` and `inputEndDateProps`? Using `Range` seems unnecessary.

Reply:
You really want to break this api don't you 😛 But yeah should be okay. I just need to remember to tell people about it :)

Reply:
Haha yeah, I'd feel worse if it weren't just one team using a "beta" component 😛